kettle/make_json.py

```diff
@@ -162,41 +164,48 @@ def get_metadata():
     return build
 
 
+def get_table(days):
+    if days:
+        return ('build_emitted_%g' % days).replace('.', '_')
+    return 'build_emitted'
+
+
 def parse_args(args):
     parser = argparse.ArgumentParser()
-    parser.add_argument('--day', action='store_true',
-                        help='Grab data for builds in the last day')
-    parser.add_argument('--days', type=float,
+    parser.add_argument('--days', type=float, default=0,
                         help='Grab data for builds within N days')
     parser.add_argument('--reset-emitted', action='store_true',
                         help='Clear list of already-emitted builds.')
     return parser.parse_args(args)
 
-def main(db, opts, outfile):
-    incremental_table = 'build_emitted'
 
+def make_rows(db, builds):
+    for rowid, path, started, finished in builds:
+        try:
+            results = db.test_results_for_build(path)
+            yield rowid, row_for_build(path, started, finished, results)
+        except IOError:
+            return
+        except:  # pylint: disable=bare-except
```

Comment:
Not super clear why this is necessary. Can we explicitly list the types of errors we are expecting and catch them?